### ISSUE TITLE
combineLastest with many SignalProducers crashes

### ISSUE THREAD

**Issue by gshahbazian** [ReactiveCocoa/ReactiveSwift#729]
To reproduce, run:

```swift
let scheduler = QueueScheduler(qos: .default, name: "RACScheduler", targeting: nil)

let signals = (0..<700).map { _ -> SignalProducer<Void, NoError> in
    return SignalProducer(value: ())
}

SignalProducer
    .combineLatest(signals)
    .start(on: scheduler)
    .logEvents()
    .start()
```

Expected: SignalProducers run and doesn't crash.
Actual: Crash with EXC_BAD_ACCESS with a huge stack trace.

Using ReactiveSwift v4.0.0

Related to these closed issues #156, https://github.com/ReactiveCocoa/ReactiveCocoa/issues/2384, https://github.com/ReactiveCocoa/ReactiveCocoa/issues/3238 but is still reproducible.

**Comment by gshahbazian** [comment]
Did some more testing and the SignalProducer doesn't need to be started on a background QueueScheduler. Just starting it will crash:

```swift
let signals = (0..<700).map { _ -> SignalProducer<Void, NoError> in
    return SignalProducer(value: ())
}

SignalProducer
    .combineLatest(signals)
    .logEvents()
    .start()
```

**Comment by leonid-s-usov** [comment]
I can't reproduce this in a Playground on the latest master. Using XCode 10.2

**Comment by leonid-s-usov** [comment]
OK so this issue, as far as I can tell right now it has nothing to do with the `CombineLatest` strategy. It never gets there.
The issue is in the implementation of this private method of the `SignalProducer`
```
private static func start<S: Sequence>(_ producers: S, _ transform: @escaping (ReversedCollection<[Signal<Value, Error>]>) -> Signal<[Value], Error>)
```

Inside, the start is called recursively for all producers in the sequence. I'll need some time to understand the idea behind, I believe there were reasons for such an implementation, but this indeed looks strange. 

```
			func start() {
				guard !producers.isEmpty else {
					transform(signals.reversed()).observe(observer)
					return
				}

				producers.removeLast().producer.startWithSignal { signal, interruptHandle in
					lifetime += interruptHandle
					signals.append(signal)

					start()
				}
			}
```

**Comment by leonid-s-usov** [comment]
I've submitted my fix at #734. 
I suspect that there may be a better version of it, will appreciate a review from people who authored the current version of the zip and combineLatest machinery. However, without further refactoring of how the Signal aggregate strategy work, I don't currently see a conceptually different way of fixing this issue

**Comment by leonid-s-usov** [comment]
Strange... is no one interested in the fix?
